packages/alchemy/src/react/types.ts

```diff
@@ -0,0 +1,26 @@
+import type { UseMutationOptions } from "@tanstack/react-query";
+
+export type BaseHookMutationArgs<
+  TData extends any = void,
+  TVariable extends any = void
+> = {
+  onSuccess?:
+    | ((data: TData, variables: TVariable, context: unknown) => unknown)
+    | undefined;
+  onError?:
+    | ((error: Error, variables: TVariable, context: unknown) => unknown)
+    | undefined;
+  onSettled?:
+    | ((
+        data: TData | undefined,
+        error: Error | null,
+        variables: TVariable,
+        context: unknown
+      ) => unknown)
+    | undefined;
+} & Partial<
+  Omit<
+    UseMutationOptions<TData, Error, TVariable, unknown>,
```

Comment:
does this not already include the `on*` methods? can this type be simplified to just the `Omit<...>`?

Reply:
Ah good catch didn't update this, but yup will change